build.gradle

```diff
@@ -471,47 +472,100 @@ project (':quasar-core') {
+                    "Implementation-Version"    :   archiveVersion.get(),
+                    "Implementation-Vendor"     :   vendor
+                )
+            }
+
+            bnd """
+Bundle-SymbolicName: co.paralleluniverse.quasar-core
+Bundle-Name: Quasar Fibers
+Bundle-Version: $version
+Import-Package: \
+ org.LatencyUtils.*;resolution:=optional,\
+ org.HdrHistogram.*;resolution:=optional,\
+ org.junit.*;resolution:=optional,\
+ *
+"""
+        }
+
+        def bundleAgentTask = tasks.register("${set.name}AgentBundle", Bundle) {
+            dependsOn shadowJarTask
+            from(zipTree(shadowJarTask.archiveFile)) {
+                include 'co/paralleluniverse/asm/**'
+                include 'co/paralleluniverse/common/asm/**'
+                include 'co/paralleluniverse/fibers/instrument/**'
+                exclude 'co/paralleluniverse/fibers/instrument/InstrumentationTask*'
+                exclude 'co/paralleluniverse/fibers/instrument/InstrumentationTKB*'
+                exclude 'co/paralleluniverse/fibers/instrument/OldSuspendablesScanner*'
+                exclude 'co/paralleluniverse/fibers/instrument/QuasarURLClassLoader*'
+                exclude 'co/paralleluniverse/fibers/instrument/SuspendablesScanner*'
+            }
+            archiveAppendix = 'osgi'
+            archiveClassifier = 'agent'
+
+            bnd """
+-include: "jar:${shadowJarTask.archiveFile.map { it.asFile.toURI() }.get()}!/META-INF/MANIFEST.MF"
+Bundle-SymbolicName: co.paralleluniverse.quasar-core.agent
+Bundle-Name: Quasar Agent
+Bundle-Version: $version
+"""
+        }
+
         def testTask = task("${set.name}Test", type: Test, dependsOn: [shadowJarTask, compileClassloadertestJava]) {
             testClassesDirs = project.sourceSets["${set.name}test"].output.classesDirs
             classpath = project.sourceSets["${set.name}test"].runtimeClasspath
             project.afterEvaluate {
-                jvmArgs "-javaagent:${shadowJarTask.archivePath}" // =vdc
+                jvmArgs "-javaagent:${shadowJarTask.archiveFile.get()}" // =vdc
             }
         }
 
         artifacts {
             archives shadowJarTask
         }
 
-        assemble.dependsOn shadowJarTask
-        build.dependsOn shadowJarTask
-        install.dependsOn shadowJarTask
-        signArchives.dependsOn shadowJarTask
-        artifactoryPublish.dependsOn shadowJarTask
+        signArchives.dependsOn shadowJarTask, bundleTask, bundleAgentTask
+        artifactoryPublish.dependsOn shadowJarTask, bundleTask, bundleAgentTask
 
-        assemble.dependsOn shadowJarTask
         test.dependsOn testTask
     }
 
-    compileJdk7Java {
-        sourceCompatibility = '1.7'
-        targetCompatibility = '1.7'
-    }
+    def jdk8ShadowJar = tasks.named('jdk8ShadowJar')
+    def jdk8Bundle = tasks.named('jdk8Bundle', Bundle)
+    def jdk8AgentBundle = tasks.named('jdk8AgentBundle', Bundle)
+
+    //    compileJdk7Java {
```

Comment:
We should delete rather than leave as commented

Reply:
Just trying to keep our `build.gradle` recognisable with the "master" copy in Ron's repo. But I can delete it anyway.